client/state/data-layer/analytics/adTracking.js

```diff
@@ -0,0 +1,23 @@
+/**
+ * External dependencies
+ */
+import { get } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { recordSelectedVertical } from 'lib/analytics/ad-tracking';
+import { SIGNUP_STEPS_SURVEY_SET } from 'state/action-types';
+
+function handleSignupStepsSurveyStart( _, action, next ) {
```

Comment:
Then we'd have `recordVerticalSelection` calling `recordSelectedVertical` :)

I could just rename this action handler `recordVerticalSelection` and then inline the contents of the original `recordVerticalSelection` function from `cliient/lib/analytics/ad-tracking.js` as per your earlier suggestion.